src/buf.rs

```diff
@@ -58,6 +63,41 @@ impl GrpcSlice {
     pub fn from_static_str(s: &'static str) -> GrpcSlice {
         GrpcSlice::from_static_slice(s.as_bytes())
     }
+
+    /// Checks whether the slice stores bytes inline.
+    pub fn is_inline(&self) -> bool {
+        self.0.refcount.is_null()
+    }
+
+    /// Reallocates current slice with given capacity.
+    ///
+    /// The length of returned slice is the exact same as given cap.
+    ///
+    /// ## Safety
+    ///
+    /// Caller is expected to initialize all available bytes to guarantee safety of this slice.
+    pub unsafe fn realloc(&mut self, cap: usize) -> &mut [MaybeUninit<u8>] {
+        if cap <= INLINED_SIZE {
+            // Only inlined slice can be reused safely.
+            if !self.0.refcount.is_null() {
+                *self = GrpcSlice::default();
+            }
+            self.0.data.inlined.length = cap as u8;
+            std::slice::from_raw_parts_mut(
+                self.0.data.inlined.bytes.as_mut_ptr() as *mut MaybeUninit<u8>,
+                cap,
+            )
+        } else {
+            *self = GrpcSlice(grpcio_sys::grpc_slice_malloc_large(cap));
```

Comment:
Seems there is still no `capacity` in `grpc_slice`?

Reply:
Indeed, `capacity == len`. That's why comment says "Caller is expected to initialize all available bytes to guarantee safety of this slice."